eng/tools/RepoTasks/CreateFrameworkListFile.cs

```diff
@@ -57,12 +57,34 @@ public override bool Execute()
                     (f.Filename.EndsWith(".dll", StringComparison.OrdinalIgnoreCase) || f.IsNative))
                 .OrderBy(f => f.Filename, StringComparer.Ordinal))
             {
-                var element = new XElement(
-                    "File",
-                    new XAttribute("Type", f.IsNative ? "Native" : "Managed"),
-                    new XAttribute(
-                        "Path",
-                        Path.Combine(f.PackagePath, f.Filename).Replace('\\', '/')));
+                string path = Path.Combine(f.PackagePath, f.Filename).Replace('\\', '/');
+                string type = f.IsNative ? "Native" : "Managed";
+                var element = new XElement("File", new XAttribute("Path", path));
+
+                if (path.StartsWith("analyzers/"))
+                {
+                    type = "Analyzer";
+
+                    if (path.EndsWith(".resources.dll"))
+                    {
+                        // omit analyzer resources
+                        continue;
+                    }
+
+                    var pathParts = path.Split('/');
+
+                    if (pathParts.Length < 3 || !pathParts[1].Equals("dotnet", StringComparison.Ordinal) || pathParts.Length > 4)
+                    {
+                        Log.LogError($"Unexpected analyzer path format {path}.  Expected  'analyzers/dotnet(/language)/analyzer.dll");
+                    }
+
+                    if (pathParts.Length > 3)
```

Comment:
Should this be
```suggestion
                    if (pathParts.Length >= 3)
```

Reply:
I specifically want it to be greater than 3.  That means there is a language directory.  If it's only 3 there is no language directory.

Reply:
```suggestion
                    // Check if we have enough parts for language directory and include it
                    if (pathParts.Length > 3)
```